File: test-fixtures/cross-language/zig/fields.zig
const Ledger = struct {
    mutableField: i32,
    immutableField: i32,
    sharedName: i32,
    pub const staticField: i32 = 1;
};

const Archive = struct {
    sharedName: i32,
};

pub fn readLedger(ledger: Ledger) i32 {
    return ledger.mutableField + Ledger.staticField;
}

File: test-fixtures/cross-language/rust/fields.rs
pub struct Ledger {
    pub mutable_field: i32,
    immutable_field: i32,
    pub shared_name: i32,
}

pub struct Archive {
    pub shared_name: i32,
}

pub enum FieldEvent {
    Changed { mutable_field: i32 },
    Indexed(i32),
}

pub fn read_ledger(ledger: &Ledger) -> i32 {
    ledger.mutable_field + ledger.shared_name
}
